index.bs

```diff
@@ -5496,20 +5534,17 @@ but MUST NOT be shorter than the longest prefix substring that satisfies the siz
 
 ### Language and Direction Encoding ### {#sctn-strings-langdir}
 
-In order to be correctly displayed in context, the language and base direction of a string [may be required](https://www.w3.org/TR/string-meta/#why-is-this-important). Strings in this API may have to be written to fixed-function [=authenticators=] and then later read back and displayed on a different platform. Thus language and direction metadata is encoded in the string itself to ensure that it is transported atomically.
-
-To encode language and direction metadata in a string that is documented as permitting it, suffix its code points with two sequences of code points:
-
-The first encodes a [=language tag=] with the code point U+E0001 followed by the ASCII values of the [=language tag=] each shifted up by U+E0000. For example, the [=language tag=] “en-US” becomes the code points U+E0001, U+E0065, U+E006E, U+E002D, U+E0055, U+E0053.
+In order to be correctly displayed in context, the language and base direction of a string [may be required](https://www.w3.org/TR/string-meta/#why-is-this-important). Strings in this API may have to be written to fixed-function [=authenticators=] and then later read back and displayed on a different platform.
 
-The second consists of a single code point which is either U+200E (“LEFT-TO-RIGHT MARK”), U+200F (“RIGHT-TO-LEFT MARK”), or U+E007F (“CANCEL TAG”). The first two can be used to indicate directionality but SHOULD only be used when neccessary to produce the correct result. (E.g. an RTL string that starts with LTR-strong characters.) The value U+E007F is a direction-agnostic indication of the end of the [=language tag=].
+For compatibility with existing fixed-function [=authenticators=] without support for dedicated language and direction metadata fields,
+Web Authentication Level 2 included provisions for embedding such metadata in the string itself to ensure that it is transported atomically.
+This encoding is NOT RECOMMENDED; [=clients=] and [=authenticators=] MAY ignore such encoding in new values.
+[=Clients=] and [=authenticators=] MAY detect and process language and direction metadata encoded in existing strings
+as described in [[webauthn-2-20210408#sctn-strings-langdir|Web Authentication Level 2 §6.4.2. Language and Direction Encoding]].
 
-So the string “حبیب الرحمان” could have two different DOMString values, depending on whether the language was encoded or not. (Since the direction is unambiguous a directionality marker is not needed in this example.)
+Instead, the Level 3 [=Web Authentication API=] provides [=resource-wide default=] fields for encoding language and direction metadata
+which SHOULD be used when supported by the [=client=] and [=authenticator=].
```

Comment:
How does a relying party know if the client/authenticator support the new language tags?

Reply:
It doesn't... but I think I see what you mean. When I wrote "SHOULD be used" I meant "[the **client**] should use them [when provided by the RP and supported by the authenticator]", but your reading was "[the **RP**] should use them [when supported by the client and authenticator]"? That's definitely an ambiguity I need to fix.

Anything else to do here after fixing that ambiguity?

Reply:
Does 5d2e401fc80c4f021d161abd23e1faafbed06768 resolve this ambiguity?